packages/astro/src/core/config/schema.ts

```diff
@@ -173,6 +173,7 @@ export const AstroConfigSchema = z.object({
 						z.literal(308),
 					]),
 					destination: z.string(),
+					priority: z.enum(['above-project', 'same-as-project', 'below-project']).optional(),
```

Comment:
I believe we should at least provide a default.

Reply:
The default is added where it is used, but I can also add it here.

 I didn't do it because then there would be a risk of the defaults becoming out of sync. If the redirect uses just the destination string instead of an object, there is no priority in the configuration, so the default has to be where it is used.

Reply:
> so the default has to be where it is used.

From this phrase, I understand "where it is used" and *this configuration* aren't the same. Did I understand correctly?

Reply:
Yes, I mean the code on `create.ts` that uses the value from the configuration to do something.

Reply:
Is the configuration ever **not** passed through this schema?

I think what @ematipico is suggesting is this:

```suggestion
					priority: z
						.enum(['above-project', 'same-as-project', 'below-project'])
						.default('above-project'),
```

This ensures `priority` is always defined and no checks need to be made later.

Reply:
Redirects can be just a string:
```ts
{
  redirects: {
    '/foo': '/bar'
  }
}
```


So the code code on `create.ts` would have to duplicate the default anyway. We transform it here... I'll send an option for that

Reply:
You could use a Zod transform to do that upfront too, iiuc!

Reply:
Using a transform now!